src/torchjd/backward/multi_task_backward.py

```diff
@@ -0,0 +1,131 @@
+        ``None``.
+    """
+    _check_optional_positive_chunk_size(parallel_chunk_size)
+
+    shared_activations = _as_tensor_list(shared_activations)
+
+    if len(shared_activations) == 0:
+        raise ValueError("`activations` cannot be an empty iterable of `Tensor`s.")
+
+    _check_retain_graph_compatible_with_chunk_size(
+        shared_activations, retain_graph, parallel_chunk_size
+    )
+
+    if len(tasks_losses) == 0:
+        raise ValueError("`tasks_losses` cannot be an empty sequence of `Tensors`s.")
+    if len(tasks_losses) != len(tasks_parameters):
+        raise ValueError("`tasks_losses` and `tasks_parameters` should have the same size.")
+
+    shared_parameters = list(shared_parameters)
+    tasks_parameters = [list(task_input) for task_input in tasks_parameters]
+
+    # Transforms that store gradient of the losses w.r.t. tasks specific parameters into their
+    # ``.grad`` fields and backpropagate the gradient of the losses w.r.t. to the shared activation.
+    task_transforms = [
+        _make_task_transform(
+            shared_activations,
+            task_parameters,
+            tensor,
+            retain_graph,
+        )
+        for task_parameters, tensor in zip(tasks_parameters, tasks_losses)
+    ]
+
+    # Transform that stacks the gradients of the losses w.r.t. activations into a Jacobian
+    stack = Stack(task_transforms)
+
+    # Transform that computes the Jacobians of the shared parameters
+    jac = Jac(shared_activations, shared_parameters, parallel_chunk_size, retain_graph)
+
+    # Transform that defines the aggregation of the jacobians into gradients
+    aggregation = make_aggregation(UnifyingStrategy(A, shared_parameters))
+
+    # Transform that stores the gradients with respect to the shared parameters
+    store = Store(shared_parameters)
+
+    backward_transform = store << aggregation << jac << stack
+
+    backward_transform(EmptyTensorDict())
+
+
+def _make_task_transform(
+    activations: list[Tensor],
+    task_parameters: list[Tensor],
+    losses: Tensor,
+    retain_graph: bool,
+) -> Transform[EmptyTensorDict, Gradients]:
+    to_differentiate = task_parameters + activations
+
+    # Transform that initialize the gradient output to 1.
+    init = Init([losses])
+
+    # Transform that computes the gradients of task losses w.r.t. the task specific parameters and
+    # the shared activation
+    grad = Grad([losses], to_differentiate, retain_graph)
+
+    # Transform that stores the task specific parameters into their ``.grad`` fields.
+    store = Store(task_parameters) << Subset(task_parameters, to_differentiate)
+
+    # Transform that stores gradient of the losses w.r.t. task specific parameters into their
+    # ``.grad`` fields and backpropagate the gradient of the losses w.r.t. to the shared activation.
+    backward_task = (Subset(activations, to_differentiate) | store) << grad << init
```

Comment:
I think we could give a name to `Subset(activations, to_differentiate)`

We used to have it like this:
```python
# Transform that selects the gradients to keep back-propagating
backpropagate = Subset(to_backpropagate, inputs)
```